### PR TITLE
lnwallet/chancloser: cooperative channel closure pkg

### PR DESCRIPTION
**Pull request by Crypt-iQ**
Introduces a new `chancloser` package and lets `peer` use it instead of `chancloser.go`. Preparatory step for the `peer` package.

### CODE REVIEW

File: lnwallet/chancloser/chancloser.go [hunk 1]
```diff
@@ -117,80 +117,78 @@ type channelCloser struct {
...
+	// We'll use this as a pivot point to ratchet our next offer up, down, or
+	// simply accept the remote party's prior offer.
 	lastFeeProposal btcutil.Amount
 
-	// priorFeeOffers is a map that keeps track of all the proposed fees
-	// that we've offered during the fee negotiation. We use this map to
-	// cut the negotiation early if the remote party ever sends an offer
-	// that we've sent in the past. Once negotiation terminates, we can
-	// extract the prior signature of our accepted offer from this map.
+	// priorFeeOffers is a map that keeps track of all the proposed fees that
```
Comment (cfromknecht) [inline lnwallet/chancloser/chancloser.go:133]: godoc formatting could be updated in a separate commit, either before or after

File: lnwallet/chancloser/chancloser.go [hunk 2]
```diff
@@ -659,62 +646,58 @@ func rachetFee(fee btcutil.Amount, up bool) btcutil.Amount {
...
-func calcCompromiseFee(chanPoint wire.OutPoint,
-	ourIdealFee, lastSentFee, remoteFee btcutil.Amount) btcutil.Amount {
+func calcCompromiseFee(chanPoint wire.OutPoint, ourIdealFee, lastSentFee,
+	remoteFee btcutil.Amount) btcutil.Amount {
 
 	// TODO(roasbeef): take in number of rounds as well?
 
-	peerLog.Infof("ChannelPoint(%v): computing fee compromise, ideal=%v, "+
-		"last_sent=%v, remote_offer=%v", chanPoint, int64(ourIdealFee),
+	chancloserLog.Infof("ChannelPoint(%v): computing fee compromise, ideal= "+
```
Comment (cfromknecht) [inline lnwallet/chancloser/chancloser.go:654]: this log will now print `ideal= lastSentFee` (extra space)

File: lnwallet/chancloser/chancloser.go [hunk 3]
```diff
@@ -245,36 +244,36 @@ func (c *channelCloser) initChanShutdown() (*lnwire.Shutdown, error) {
...
-	// If we attempt to shutdown the channel for the first time, and we're
-	// not in the closeIdle state, then the caller made an error.
+func (c *ChanCloser) ShutdownChan() (*lnwire.Shutdown, error) {
+	// If we attempt to shutdown the channel for the first time, and we're not
+	// in the closeIdle state, then the caller made an error.
 	if c.state != closeIdle {
 		return nil, ErrChanAlreadyClosing
 	}
 
-	peerLog.Infof("ChannelPoint(%v): initiating shutdown of", c.chanPoint)
```
Comment (cfromknecht) [inline lnwallet/chancloser/chancloser.go]: log message modified during code move

File: lnwallet/chancloser/chancloser.go [hunk 4]
```diff
@@ -340,72 +338,74 @@ func maybeMatchScript(disconnect func() error,
...
+			shutdownMsg.Address,
 		); err != nil {
 			return nil, false, err
 		}
 
 		// Once we have checked that the other party has not violated option
 		// upfront shutdown we set their preference for delivery address. We'll
 		// use this when we craft the closure transaction.
-		c.remoteDeliveryScript = shutDownMsg.Address
+		c.remoteDeliveryScript = shutdownMsg.Address
```
Comment (cfromknecht) [inline lnwallet/chancloser/chancloser.go:386]: variable name changed during code move (few other places as well)

File: lnwallet/chancloser/chancloser.go [hunk 5]
```diff
@@ -340,72 +338,74 @@ func maybeMatchScript(disconnect func() error,
...
-		// the wire.
+		// We'll generate a shutdown message of our own to send across the
+		// wire.
 		localShutdown, err := c.initChanShutdown()
 		if err != nil {
 			return nil, false, err
 		}
 
-		peerLog.Infof("ChannelPoint(%v): Responding to shutdown",
+		chancloserLog.Infof("ChannelPoint(%v): responding to shutdown",
```
Comment (cfromknecht) [inline lnwallet/chancloser/chancloser.go:395]: log message changed during code move

**Comment by Crypt-iQ** [comment]
Is the btcd integration test failure a flake? It seems something isn't added to the whitelist when it should be. I'm not sure how to handle that.

**Comment by cfromknecht** [comment]
@Crypt-iQ yeah just a flake, I wouldn't worry about it the required checks are already green

**Comment by cfromknecht** [comment]
yeah package placement is definitely up for debate, just seemed like a the best places atm since it mirrors `chanfunding` in a way
